docs/reference/contracts/rate-providers.md

```diff
@@ -39,3 +39,132 @@ This contract makes use of Chainlink's registry contract so it can handle if Cha
+
+\*Bear in mind that the tokens used for demonstration in these examples all have 18 decimals and Balancer natively uses 18 decimals for internal accounting. If tokens have different decimals, the scaled balances scale with the tokens decimals as well.
+
+After the token balances have been upscaled, they are fed into `_calcOutGivenIn` [here](https://dashboard.tenderly.co/tx/mainnet/0x72d756d0fcd663343ca1b2adcfc7e114e8598bc0be28386752f16222384a29b3?trace=0.4.2.2.19.11.0.3.0.5).
+That is how `rateProvider` feeds a rate into the pricing equation for ComposableStablePools.
+
+### Implementation in a MetaStablePool 
+How scaling is used can be seen in the below code snippet.
+
+```
+function _scalingFactor(IERC20 token) internal view virtual override returns (uint256) {
+        uint256 baseScalingFactor = super._scalingFactor(token);
+        uint256 priceRate = _priceRate(token);
+        // Given there is no generic direction for this rounding, it simply follows the same strategy as the BasePool.
+        return baseScalingFactor.mulDown(priceRate);
+    }
+```
+
+### Example for MetaStablePools (superseded by ComposableStablePool) swap utilising rate Providers.
+
+Looking at this [transaction](https://etherscan.io/tx/0x67f477517acf6e0c91ec7997e665ca25d2806da060af30272876742584f0aa21). 50 ETH is being exchanged for 46.68 rETH. This pool is a MetaStablePool. According to the next table the rate, the `rateProvider`  supplies is being taken into account during the swap. Looking at the Trade equations, a well suited parameter to weave in the `rate` is the balances which are used to compute `OutGivenIn` or `InGivenOut`.
+
+Querying the balances of this pool via `vault.getPoolTokens(poolId)` returns  
+20040415915824227571764 for rETH and 21953505292747563228232 for WETH. 
+  
+| Token                                             | balances                              |  rate                |  scaled balance                           | 
+| -----------                                       | -----------                           |  -----------         |  -----------                              | 
+| rEth 0xae78736cd615f374d3085123a210448e74fc6393   | 20040415915824227571764                | 1070121751154609309  |  21445684973708525874136                  | 
+| WETH 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2   | 21953505292747563228232                | 1000000000000000000  |  21953505292747563228232                  | 
+
+The token balances used in the Trade Equations are then [upscaled](https://dashboard.tenderly.co/tx/mainnet/0x67f477517acf6e0c91ec7997e665ca25d2806da060af30272876742584f0aa21?trace=0.5.2.1.5.11) the scaled balance. Only after upscaling are the apparent balances supplied to
+[`OnSwapGivenIn()`](https://dashboard.tenderly.co/tx/mainnet/0x67f477517acf6e0c91ec7997e665ca25d2806da060af30272876742584f0aa21?trace=0.5.2.1.5.13.2). 
+
+
+### Rate Provider being used in a MetaStablePool swap
```

Comment:
They are similar in content but the difference is that they use different scaling implementations. I think it can be helpful for people to understand how balancer does scaling with rateProviders for these two pool types as they are heavily used. Not sure what you are referring to tbh.